File: VB/WinWebSolution.Module/Controllers/PhoneNumberNestedListViewController.vb
Imports System
Imports System.Reflection
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.BaseImpl

Namespace WinWebSolution.Module
    Public Class PhoneNumberNestedListViewController
        Inherits NestedListViewControllerBase

        Private saMasterDetailViewInfoAction As SimpleAction = Nothing
        Public Sub New()
            TargetObjectType = GetType(PhoneNumber)
            saMasterDetailViewInfoAction = New SimpleAction(Me, "MasterDetailViewInfoAction", DevExpress.Persistent.Base.PredefinedCategory.View)
            AddHandler saMasterDetailViewInfoAction.Execute, AddressOf saMasterDetailViewInfoAction_Execute
        End Sub
        Private Sub saMasterDetailViewInfoAction_Execute(ByVal sender As Object, ByVal e As SimpleActionExecuteEventArgs)
            MasterDetailViewInfo()
        End Sub
        Private Sub MasterDetailViewInfo()
            Console.WriteLine(String.Format("MasterViewId = '{0}'", Me.MasterDetailViewId))
        End Sub
        Protected Overrides Sub OnActivated()
            MyBase.OnActivated()
            AddHandler View.ControlsCreated, AddressOf View_ControlsCreated
        End Sub
        Private Sub View_ControlsCreated(ByVal sender As Object, ByVal e As EventArgs)
            MasterDetailViewInfo()
        End Sub
    End Class
End Namespace

File: CS/WinWebSolution.Module/Controllers/PhoneNumberNestedListViewController.cs
using System;
using System.Reflection;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.BaseImpl;

namespace WinWebSolution.Module {
    public class PhoneNumberNestedListViewController : NestedListViewControllerBase {
        private SimpleAction saMasterDetailViewInfoAction = null;
        public PhoneNumberNestedListViewController() {
            TargetObjectType = typeof(PhoneNumber);
            saMasterDetailViewInfoAction = new SimpleAction(this, "MasterDetailViewInfoAction", DevExpress.Persistent.Base.PredefinedCategory.View);
            saMasterDetailViewInfoAction.Execute += new SimpleActionExecuteEventHandler(saMasterDetailViewInfoAction_Execute);
        }
        void saMasterDetailViewInfoAction_Execute(object sender, SimpleActionExecuteEventArgs e) {
            MasterDetailViewInfo();
        }
        private void MasterDetailViewInfo() {
            Console.WriteLine(string.Format("MasterViewId = '{0}'", this.MasterDetailViewId));
        }
        protected override void OnActivated() {
            base.OnActivated();
            View.ControlsCreated += new EventHandler(View_ControlsCreated);
        }
        void View_ControlsCreated(object sender, EventArgs e) {
            MasterDetailViewInfo();
        }
    }
}
